components/info_box/ActionBox.tsx

```diff
@@ -0,0 +1,222 @@
+"use client";
+
+import InfoBox from "./InfoBox";
+import { rubik } from "@/utils/Fonts";
+
+import { Form, Button, FormCheck, Row, Col, InputGroup } from "react-bootstrap";
+
+import LocalTeamBox from "./LocalTeamBox";
+import { submitANForm } from "@/utils/AnApiSubmission";
+import FormCheckInput from "react-bootstrap/esm/FormCheckInput";
+import FormCheckLabel from "react-bootstrap/esm/FormCheckLabel";
+import { useRef, useState, useEffect } from "react";
+
+const emailErrorToErrorMessage = (code: EmailErrorCode) => {
+  switch (code) {
+    case "EMAIL_INVALID":
+      return "Please add a valid email address.";
+    case "SERVER_ERROR":
+      return "Something went wrong signing you up. Please try again?";
+    default:
+      return "";
+  }
+};
+
+const checkErrorToErrorMessage = (code: "MISSING_CONSENT" | null) => {
+  switch (code) {
+    case "MISSING_CONSENT":
+      return "Please tick the signup box.";
+    default:
+      return "";
+  }
+};
+
+type FormData = {
+  emailOptIn: boolean;
+  email: string;
+};
+
+const initialFormState: FormData = {
+  emailOptIn: false,
+  email: "",
+};
+
+const ActionBox = ({
```

Comment:
Have a suspicion that at some point we'll want to factor out just the subscribe form (possibly with some parameterised copy) to re-use elsewhere. Don't suggest tackling that now though.

Also obvs a fair chunk of shared logic between this component & the postcode lookup, but they can't be the same thing. At some point worth looking at any common functions that can be factored out & shared, but think we'll always need two top-level components for the 2 forms.

Anyway, just thoughts for future, in no way urgent now.

Reply:
Yeah I agree, factoring out the localStorage logic seems worth doing but at that point I start to think we should pull in a localStorage hook from a library.